packages/commonwealth/client/scripts/helpers/validateJson.ts

```diff
@@ -0,0 +1,159 @@
+          "type": "object",
+          "properties": {
+            "field_name": { "type": "string" }
+          },
+          "required": [
+            "field_name"
+          ],
+          "additionalProperties": false
+        }
+      },
+      "required": [ "divider" ],
+      "additionalProperties": false
+    },
+    "text": {
+      "type": "object",
+      "properties": {
+        "text": {
+          "type": "object",
+          "properties": {
+            "field_name": { "type": "string" },
+            "field_value": { "type": "string" },
+            "field_type": {
+              "type": "string",
+              "enum": [
+                "h1",
+                "h2",
+                "h3",
+                "text"
+              ]
+            }
+          },
+          "required": [
+            "field_name",
+            "field_value"
+          ],
+          "additionalProperties": false
+        }
+      },
+      "required": [ "text" ],
+      "additionalProperties": false
+    },
+    "input": {
+      "type": "object",
+      "properties": {
+        "input": {
+          "type": "object",
+          "properties": {
+            "field_name": { "type": "string" },
+            "field_label": { "type": "string" },
+            "field_value": { "type": "string" },
+            "field_ref": { "type": "string" },
+            "formatter": { "$ref": "#/$defs/formatter" }
+          },
+          "required": [
+            "field_name",
+            "field_label",
+            "field_ref"
+          ],
+          "additionalProperties": false
+        }
+      },
+      "required": [ "input" ],
+      "additionalProperties": false
+    },
+    "dropdown": {
+      "type": "object",
+      "properties": {
+        "dropdown": {
+          "type": "object",
+          "properties": {
+            "field_name": { "type": "string" },
+            "field_label": { "type": "string" },
+            "field_value": { "type": "string" },
+            "field_ref": { "type": "string" },
+            "formatter": { "$ref": "#/$defs/formatter" },
+            "field_options": {
+              "type": "array",
+              "items": {
+                "type": "object",
+                "properties": {
+                  "label": {
+                    "type": "string"
+                  },
+                  "value": {
+                    "type": "string"
+                  }
+                }
+              }
+            }
+          },
+          "required": [
+            "field_name",
+            "field_label",
+            "field_ref",
+            "field_options"
+          ],
+          "additionalProperties": false
+        }
+      },
+      "required": [ "dropdown" ],
+      "additionalProperties": false
+    }
+  }
+};
+
+type AnyKeyAnyValue = Record<string, any>;
+
+export default function isValidJson(data: AnyKeyAnyValue): boolean {
+  const validate = validator(JSON.parse(JSON.stringify(schema)))
+  return validate(data);
```

Comment:
shouldn't we return some errors (if applicable) along with the boolean?

Reply:
We should yes, but I would do that outside of this function. If this function returns false, then render the error. This package only can only return true or false, is it valid or not. If we want to give more granular responses we would need to know all the error messages and then write some more logic to get this information to the user.